ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/StreamAggregateBuilderTest.java

```diff
@@ -420,14 +438,17 @@ public void shouldBuildTumblingWindowedAggregateCorrectly() {
 
     // Then:
     assertThat(result.getTable(), is(windowedWithWindowBounds));
+    verify(gracePeriodClause).toDuration();
+    verify(retentionClause).toDuration();
```

Comment:
not sure how much extra values these assertions give and they force us to use spys. is there any way we can check that the properties are being set on the table instead?

Reply:
Like I mentioned.. KTable does not allow me to read these properties out.. So had the spys ensure if grace period and retention are set, then some code uses them as Duration..